src/pages/slots/CreateSlot.tsx

```diff
@@ -0,0 +1,63 @@
+import { MainContainer, Label, Input, Button, Select, InputContainer, EditContainer, EditCapacity, TitleContainer, Title, ScreenContainer } from "./CreateSlot.styles";
+import EditIcon from "../../assets/edit-icon.png";
+import EditSlotForm from "./forms/EditCapacityForm";
+import { useRef, useState } from "react";
+import type { ReactNode } from "react";
+import AddIcon from "../../assets/add-icon.svg";
+import Modal from "../../components/modal/Modal";
+import BackIcon from "../../assets/back-icon.png";
+import { useNavigate } from "react-router-dom";
+
+function CreateSlot() {
+
+
+
+    const DaysOfWeek = ["Lunes", "Martes", "Miércoles", "Jueves", "Viernes", "Sábado", "Domingo"];
+
+    const [modal, setModal] = useState<ReactNode | null>(null);
```

Comment:
Este useState está raro... Declarás que el tipo de dato de este useState va a ser un `ReactNode | null`, pero cuando cerras el modal la accion de `onClose` lo que hace es setear el valor en `false` el cual es un boolean, además, cuando haces click en el `EditCapacity` lo que estas haciendo es setear el modal con el componente del formulario, pero despues ese dato no lo usas para nada (solo para chequear si hay que mostrar el modal o no).
Para eso es mejor que el useState directamente tenga el tipo de dato boolean

```suggestion
    const [modal, setModal] = useState<boolean>(false);
```